addons/website_livechat/static/src/components/discuss/discuss_tests.js

```diff
@@ -0,0 +1,308 @@
+        channel_type: 'livechat',
+        id: 11,
+        livechat_operator_id: this.data.currentPartnerId,
+        livechat_visitor_id: 11,
+        members: [this.data.currentPartnerId, this.data.publicPartnerId],
+    });
+    await this.start();
+    await afterNextRender(() => {
+        document.querySelector(`
+            .o_DiscussSidebarItem[data-thread-local-id="${
+                this.env.models['mail.thread'].findFromIdentifyingData({
+                    id: 11,
+                    model: 'mail.channel',
+                }).localId
+            }"]
+        `).click();
+    });
+
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner',
+        "should have a visitor banner",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner_avatar',
+        "should show the visitor avatar in the banner",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_VisitorBanner_avatar').dataset.src,
+        "/mail/static/src/img/smiley/avatar.jpg",
+        "should show the default avatar",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner_onlineStatusIcon',
+        "should show the visitor online status icon on the avatar",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_VisitorBanner_country').dataset.src,
+        "/base/static/img/country_flags/FAKE.png",
+        "should show the flag of the country of the visitor",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner_visitor',
+        "should show the visitor name in the banner",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_VisitorBanner_visitor').textContent,
+        "Visitor #11",
+        "should have 'Visitor #11' as visitor name",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner_language',
+        "should show the visitor language in the banner",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_VisitorBanner_language').textContent,
+        "English",
+        "should have 'English' as language of the visitor",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner_website',
+        "should show the visitor website in the banner",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_VisitorBanner_website').textContent,
+        "General website",
+        "should have 'General website' as website of the visitor",
+    );
+    assert.containsOnce(
+        document.body,
+        '.o_VisitorBanner_history',
+        "should show the visitor history in the banner",
+    );
+    assert.strictEqual(
+        document.querySelector('.o_VisitorBanner_history').textContent,
+        "Home → Contact",
+        "should have 'Home → Contact' as history of the visitor",
+    );
+});
+
+QUnit.test('livechat with logged visitor should show visitor banner', async function (assert) {
```

Comment:
Would expect a single assertion to check there's a visitor banner.
Same with previous test.
Prefer dedicated test for the rendering of visitor banner.
If there are differences between logged and non-logged visitors, again dedicated tests for that.